src/api/remote.rs

```diff
@@ -128,6 +143,18 @@ pub struct Integration {
     #[getset(get = "pub")]
     #[builder(setter(into))]
     protocol: Protocol,
+
+    /// Specifies if we want to scan specific branches
+    #[getset(get = "pub")]
+    import_branches: bool,
+
+    /// Specifies if we want to scan specific tags
+    #[getset(get = "pub")]
+    import_tags: bool,
```

Comment:
This may make marshalling/unmarshalling data from the database harder, so maybe don't bother with this suggestion if it is, but I'd recommend having an `imports` field like this:
```rust
imports: HashSet<GitImports>
```
Where `GitImports` is something like:
```rust
enum GitImports {
  Tags,
  Branches
}
```

Currently your description of what the boolean is is stored in the name of the field (`import_tags`, `import_branches`). The problem with this is that if the value ever becomes detached from the name, it's meaning is lost. We can use enums to make data that is the same shape as a boolean but in a way where the value itself describes its meaning. As a bonus, it's now trivial to add additional states by adding new constructors to the enum and the type system can prevent accidental use of the wrong boolean value.

Even if you don't implement this suggestion, the general term for losing information because of booleans is called _boolean blindness_. This [article](https://blakesmith.me/2019/05/07/rust-patterns-enums-instead-of-booleans.html) talks about it for Rust, but it's also something worth thinking about in other languages where it can be avoided.

Reply:
It looks like you may be able to use `git::Reference` actually.